File: Scripts/volatile_and_force_compiler_task/palindrome-detection/perl/palindrome-detection.pl
#!/usr/bin/perl

my $r = 1;

sub executeTask {
	my ($i) = @_;

	my $a = "saippuakivikauppias" + $i;
	chomp ( $palindrome = $a );

	@palindrome = split( //, $palindrome );

	@backwards = reverse( @palindrome );

	$sizeOfPalindrome = @palindrome;

	for( $i = 0; $i < $sizeOfPalindrome; $i++ ){
		if( @palindrome[$i] ne @backwards[$i] ){
			$pali = "";
			last;
		}
		else{
			$pali = "true";
		}
		
	} 
	return ($i);
}


for ( $b=0; $b < 2000000000; $b = $b + 1 ) {
	$r = executeTask($b);
}

File: scripts/task_tested_with_compile_script/palindrome-detection/c/palindrome-detection.c
#include <string.h>

int palindrome(const char*);

int executeTask(int i) {
	char* array = "saippuakivikauppias" + (char)i;
	palindrome(array);

   return i;
}

int main() {

   	volatile int r = 1;
	for (int i=0; i<2000000000; ++i) {
		r = executeTask(i);
	}
	return 0;
}

int palindrome(const char *s)
{
   int l;
   l = strlen(s);
   for(int i=0; i<l/2; i++)
   {
     if ( s[i] != s[l-i-1] ) return 0;
   }
   return 1;
}
